client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -128,7 +128,11 @@ public WriteBufferManager(
     super(taskMemoryManager, taskMemoryManager.pageSizeBytes(), MemoryMode.ON_HEAP);
     this.bufferSize = bufferManagerOptions.getBufferSize();
     this.spillSize = bufferManagerOptions.getBufferSpillThreshold();
-    this.instance = serializer.newInstance();
+    // columnar shuffle reader use the serialized data directly
+    if (serializer != null) {
```

Comment:
@loukey-lj In writer side the serializer is specify with null directly in gluten side code.  In reader side  it's taken from `rssHandle.getDependency().serializer()`.  Sure serializeStream method is not implemented, because serialize work is done in native engine layer